pkg/logs/tag/provider.go

```diff
@@ -28,32 +30,70 @@ var NoopProvider Provider = &noopProvider{}
 
 // provider caches a list of up-to-date tags for a given entity polling periodically the tagger.
 type provider struct {
-	entityID string
-	tags     []string
-	done     chan struct{}
-	mu       sync.Mutex
+	entityID             string
+	tags                 []string
+	done                 chan struct{}
+	forceRefresh         bool
+	forceRefreshDuration time.Duration
+	taggerWarmupDuration time.Duration
+	mu                   sync.Mutex
 }
 
 // NewProvider returns a new Provider.
 func NewProvider(entityID string) Provider {
-	return &provider{
-		entityID: entityID,
-		tags:     []string{},
-		done:     make(chan struct{}),
+	p := &provider{
+		entityID:     entityID,
+		tags:         []string{},
+		done:         make(chan struct{}),
+		forceRefresh: config.Datadog.GetBool("logs_config.k8s_wait_for_tags"),
 	}
+	if p.forceRefresh {
+		p.forceRefreshDuration = config.Datadog.GetDuration("logs_config.force_tagger_call_duration") * time.Second
+		p.taggerWarmupDuration = config.Datadog.GetDuration("logs_config.tagger_warmup_duration") * time.Second
+	}
+	return p
 }
 
 // GetTags returns the list of up-to-date tags.
 func (p *provider) GetTags() []string {
+	if p.getForceRefresh() {
+		p.updateTags()
```

Comment:
how costly is a call to `updateTags`? If it's a little costly and we're handling a lot of log entries in the first few seconds, calling this for every new log entry could quickly become very costly.

Reply:
`updateTags` calls the tagger and the tagger itself uses the `cache` pkg to cache pods

Reply:
in case of cache miss, the tagger queries the kubelet (users can enable a retry duration until the tagger find the container in the kubelet) but it's zero by default. once the entity is found in the tagger all the next tagger calls for log entries will hit the cache

Reply:
ok, thanks for the explanation. I'm surprised there's a cache here if the tagger has its own cache (and that cache is relatively efficient already). Might be interesting to dig into why this cache was added in the first place

Reply:
To be more precise the cache is used by the PodWatcher - called by the kubelet tag collectors. But yes I agree why this logs tag provider doesn't always query the tagger. Maybe @DataDog/logs-intake can answer this

Reply:
removed the local tags cache, as discussed with @ajacquemot